hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/utils/DeploymentHelper.java

```diff
@@ -101,13 +106,20 @@ public static void successCancellation(final JpaAction action, final ActionRepos
      *
      * @return the result of the action
      */
-    public static <T> T runInNewTransaction(@NotNull final PlatformTransactionManager txManager,
+    public static <T> Optional<T> runInNewTransaction(@NotNull final PlatformTransactionManager txManager,
             final String transactionName, final int isolationLevel, @NotNull final TransactionCallback<T> action) {
         final DefaultTransactionDefinition def = new DefaultTransactionDefinition();
         def.setName(transactionName);
         def.setReadOnly(false);
         def.setPropagationBehavior(TransactionDefinition.PROPAGATION_REQUIRES_NEW);
         def.setIsolationLevel(isolationLevel);
-        return new TransactionTemplate(txManager, def).execute(action);
+        final TransactionTemplate transactionTemplate = new TransactionTemplate(txManager, def);
+
+        try {
+            return Optional.of(transactionTemplate.execute(action));
```

Comment:
I guess we need to use Optional.ofNullable here as we don't know what we get back from the action. Additionally, I would rather catch the Runtime exception and log the cause if it is present as well

Reply:
good catch about the `Optional.ofNullable` I adapted it :wink: 
Regarding the RuntimeExceptoin, it is caught already with the `catch (final Exception e)` or am I missing something?

Reply:
Yes, you are right, however we can be more restrictive and catch the `RuntimeException`, btw. `TransactionException` is also a `RuntimeException`. I can't imagine receiving some other form of `Exception` here. Nevertheless, could you please also log the cause if present in either case